tests/test_queuing.py

```diff
@@ -0,0 +1,385 @@
+
+            assert len(mock_queuing_context.queue) == 1
+            assert op in mock_queuing_context.queue
+
+            QueuingContext.remove(op)
+
+        assert not mock_queuing_context.queue
+
+    def test_remove_operator_not_in_list(self, mock_queuing_context):
+        """Test that remove_operator does not fail when the operator to be removed is not in the queue."""
+
+        op1 = qml.PauliZ(0)
+        op2 = qml.PauliZ(1)
+        assert not mock_queuing_context.queue
+
+        with mock_queuing_context:
+            QueuingContext.append(op1)
+
+            assert len(mock_queuing_context.queue) == 1
+            assert op1 in mock_queuing_context.queue
+
+            QueuingContext.remove(op2)
+
+        assert len(mock_queuing_context.queue) == 1
+        assert op1 in mock_queuing_context.queue
+
+    def test_append_multiple_queues(self, three_mock_queuing_contexts):
+        """Test that append appends the operator to multiple queues."""
+
+        op = qml.PauliZ(0)
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+        with three_mock_queuing_contexts[0]:
+            with three_mock_queuing_contexts[1]:
+                with three_mock_queuing_contexts[2]:
+                    QueuingContext.append(op)
+
+        assert len(three_mock_queuing_contexts[0].queue) == 1
+        assert op in three_mock_queuing_contexts[0].queue
+
+        assert len(three_mock_queuing_contexts[1].queue) == 1
+        assert op in three_mock_queuing_contexts[1].queue
+
+        assert len(three_mock_queuing_contexts[1].queue) == 1
+        assert op in three_mock_queuing_contexts[1].queue
+
+    def test_remove_operator_multiple_queues(self, three_mock_queuing_contexts):
+        """Test that remove_operator removes the operator from the queue."""
+
+        op = qml.PauliZ(0)
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+        with three_mock_queuing_contexts[0]:
+            with three_mock_queuing_contexts[1]:
+                with three_mock_queuing_contexts[2]:
+                    QueuingContext.append(op)
+
+                    assert len(three_mock_queuing_contexts[0].queue) == 1
+                    assert op in three_mock_queuing_contexts[0].queue
+
+                    assert len(three_mock_queuing_contexts[1].queue) == 1
+                    assert op in three_mock_queuing_contexts[1].queue
+
+                    assert len(three_mock_queuing_contexts[2].queue) == 1
+                    assert op in three_mock_queuing_contexts[2].queue
+
+                    QueuingContext.remove(op)
+
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+
+class TestQueue:
```

Comment:
This set of tests is new in this PR

Reply:
I feel like these tests are functionally the same as the tests above (just broader in scope since arbitrary objects are queued), since tests on the mocked QueuingContext are essentially just testing the Queue class.

Reply:
I suppose the tests above are more like context integration tests, and the tests below are general integration tests, but there is still a lot of overlap

Reply:
Yes, but the tests in `Queue` will not actually by passed by `QueuingContext` since they all involve abstract methods :laughing:

Reply:
With the new ``Queue`` class, the need for that mock object (which is actually just inheriting from ``QueuingContext`` but done via monkeypatching) doesn't seem necessary. I've removed some of the tests from ``TestQueuingContext`` which were duplicated by the new ``TestQueue`` tests